scylla/src/transport/topology.rs

```diff
@@ -380,6 +384,7 @@ impl MetadataReader {
     /// Creates new MetadataReader, which connects to initially_known_peers in the background
     #[allow(clippy::too_many_arguments)]
     pub(crate) fn new(
+        initial_known_nodes: Vec<KnownNode>,
         initially_known_peers: Vec<ResolvedContactPoint>,
```

Comment:
What is the difference between `initial_known_nodes` and `initially_known_peers`? Their names are very similar.

Reply:
Perhaps a better naming could be: `initial_nodes_unresolved` and `initial_nodes_resolved`. WDYT?

Reply:
It's strange that we need to pass both resolved and unresolved nodes, it sounds redundant. Why can't `MetadataReader` do the resolution itself?

Reply:
We resolve early to know that we have at least one succesfully resolved address upon creating `Session`.

Reply:
How is that in conflict with resolving addresses in `MetadataReader::new` or `MetadataReader::read_metadata`? Both functions are called in the process of creating a `Session`.

Reply:
That's right, we can alter `Cluster::new()` return type from `Result<_, QueryError>` to `Result<_, NewSessionError>` and then we can indeed resolve hostnames in `MetadataReader::new()`. Good idea!